### PR TITLE
common: removed unnecessary kerl_init.

### PR DESCRIPTION
**Pull request by oopsmonk**
This PR removed unnecessary kerl_init & kerl_reset calls, and make sure kerl has initialized before calling it. 

# Test Plan:
CI pass

### CODE REVIEW

File: common/helpers/sign.c [hunk 1]
```diff
@@ -133,8 +131,6 @@ IOTA_EXPORT flex_trit_t* iota_flex_sign_address_gen(
   iss_kerl_key_digest(key, key, key_length, &kerl);
   iss_kerl_address(key, key, security * HASH_LENGTH_TRIT, &kerl);
 
-  kerl_reset(&kerl);
```
Comment (th0br0) [inline common/helpers/sign.c]: These resets are needed to make sure that we don't leak key data.

File: common/helpers/sign.c [hunk 2]
```diff
@@ -49,8 +49,6 @@ IOTA_EXPORT char* iota_sign_address_gen(char const* const seed,
...
```
Comment (th0br0) [inline common/helpers/sign.c]: Needed for security paranoia reasons

File: common/helpers/checksum.c [hunk 3]
```diff
@@ -50,7 +49,6 @@ IOTA_EXPORT flex_trit_t* iota_flex_checksum(const flex_trit_t* flex_trits,
                                             const size_t num_trits,
                                             const size_t checksum_length) {
   Kerl kerl;
-  init_kerl(&kerl);
```
Comment (th0br0) [inline common/helpers/checksum.c:53]: `kerl_hash` doesn't init the instance, so this is required

File: common/helpers/checksum.c [hunk 4]
```diff
@@ -20,7 +20,6 @@
 IOTA_EXPORT char* iota_checksum(const char* input, const size_t input_length,
                                 const size_t checksum_length) {
   Kerl kerl;
-  init_kerl(&kerl);
```
Comment (th0br0) [inline common/helpers/checksum.c:23]: kerl_hash doesn't init, so this is required

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: common/helpers/checksum.c (hunk 4 above)
Comment (oopsmonk) [reply common/helpers/checksum.c:23]: `init_kerl` is moved into `kerl_hash` makes the initialization stay close to absorb/squeeze if possible.

File: common/helpers/checksum.c (hunk 3 above)
Comment (oopsmonk) [reply common/helpers/checksum.c:53]: `init_kerl` is moved into `kerl_hash` makes the initialization stay close to absorb/squeeze if possible.

File: common/helpers/sign.c (hunk 1 above)
Comment (oopsmonk) [reply common/helpers/sign.c]: kerl is a local object here that should not have leaking problem.

File: common/helpers/sign.c (hunk 2 above)
Comment (oopsmonk) [reply common/helpers/sign.c]: kerl is a local object here that should not have leaking problem.

File: common/helpers/checksum.c (hunk 4 above)
Comment (th0br0) [reply common/helpers/checksum.c:23]: you're right - didn't notice that change

File: common/helpers/sign.c (hunk 1 above)
Comment (th0br0) [reply common/helpers/sign.c]: It's about cleaning up the memory after storing secrets in it (zeroing it out)

Comment (oopsmonk) [reply common/helpers/sign.c]: right, sorry. btw should we zero out signature too?

Comment (th0br0) [reply common/helpers/sign.c]: no, the signature is public anyway, so we don't need to worry about that one.

**Review by th0br0 — APPROVED** [review]

**Pull request merged**